File: WebsiteHoTroDichVuBanDoc_Backend/WebsiteHoTroDichVuBanDoc_Backend/app/models/phong.py
from typing import Optional, Dict, Any
from app.models.db_base import DBModel

class PhongBase(DBModel):
    tenPhong: str
    loaiPhong: Optional[str] = None
    sucChua: Optional[int] = None
    viTri: Optional[str] = None
    trangThai: str = 'moCua'
    thongTinThietBiTaiPhong: Optional[Dict[str, Any]] = None
    ghiChu: Optional[str] = None

class PhongCreate(PhongBase):
    pass

class PhongUpdate(DBModel):
    tenPhong: Optional[str] = None
    loaiPhong: Optional[str] = None
    sucChua: Optional[int] = None
    viTri: Optional[str] = None
    trangThai: Optional[str] = None
    thongTinThietBiTaiPhong: Optional[Dict[str, Any]] = None
    ghiChu: Optional[str] = None

class Phong(PhongBase):
    maPhong: int

File: WebsiteHoTroDichVuBanDoc/Backend/src/main/java/com/library/website/models/Phong.java
package com.library.website.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "phong", schema = "public")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Phong {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "maphong")
    private Long maPhong;

    @Column(name = "tenphong", nullable = false)
    private String tenPhong;

    @Column(name = "loaiphong")
    private String loaiPhong;

    @Column(name = "succhua")
    private Integer sucChua;

    @Column(name = "vitri")
    private String viTri;

    @Column(name = "trangthai")
    private String trangThai = "moCua";

    @Column(name = "thongtinthietbitaiphong", columnDefinition = "jsonb")
    private String thongTinThietBiTaiPhong;

    @Column(name = "ghichu")
    private String ghiChu;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
